src/overlays/actors/ovl_En_Guruguru/z_en_guruguru.c

```diff
@@ -21,32 +45,356 @@ const ActorInit En_Guruguru_InitVars = {
+            func_80151938(globalCtx, textIDs[this->textIdIndex]);
+            return;
+        }
+        func_801A3B48(0);
+        func_80151BB4(globalCtx, 0x13);
+        func_80BC6E10(this);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Guruguru_0x80BC6BF0/func_80BC6D64.asm")
+void func_80BC73F4(EnGuruguru* this) {
+    func_801A3B48(0);
+    this->unk268 = 1;
+    this->headZRotTarget = 0;
+    this->unk272 = 2;
+    this->actionFunc = func_80BC7440;
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Guruguru_0x80BC6BF0/func_80BC6E00.asm")
+void func_80BC7440(EnGuruguru* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        this->actor.parent = NULL;
+        this->textIdIndex++;
+        this->actor.textId = textIDs[this->textIdIndex];
+        func_801A3B48(1);
+        func_800B8500(&this->actor, globalCtx, 400.0f, 400.0f, -1);
+        this->unk268 = 0;
+        gSaveContext.weekEventReg[38] |= 0x40;
+        this->actionFunc = func_80BC7520;
+    } else {
+        func_800B8A1C(&this->actor, globalCtx, GI_MASK_BREMEN, 300.0f, 300.0f);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Guruguru_0x80BC6BF0/func_80BC6E10.asm")
+void func_80BC7520(EnGuruguru* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        this->actionFunc = func_80BC7068;
+    } else {
+        func_800B8500(&this->actor, globalCtx, 400.0f, 400.0f, -1);
+    }
+}
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Guruguru_0x80BC6BF0/func_80BC6F14.asm")
+void EnGuruguru_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnGuruguru* this = THIS;
+    s32 yaw;
+    Player* player = PLAYER;
+    s16 yawTemp;
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Guruguru_0x80BC6BF0/func_80BC701C.asm")
+    if (!gSaveContext.isNight) {
+        if (this->actor.params == 1) {
+            Actor_MarkForDeath(&this->actor);
+            return;
+        }
+    } else if (this->actor.params == 0 || this->actor.params == 2) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Guruguru_0x80BC6BF0/func_80BC7068.asm")
+    this->actionFunc(this, globalCtx);
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Guruguru_0x80BC6BF0/func_80BC73F4.asm")
+    if (this->actor.params == 2) {
+        if (fabsf(player->actor.world.pos.y - this->actor.world.pos.y) < 100.0f) {
+            func_801A1DB8(&this->actor.projectedPos, 0x2E, 540.0f);
+        }
+        return;
+    }
+    if (fabsf(player->actor.world.pos.y - this->actor.world.pos.y) < 200.0f) {
+        func_801A1D44(&this->actor.projectedPos, 0x2E, 540.0f);
+    }
+    if (this->unusedTimer != 0) {
+        this->unusedTimer--;
+    }
+    if (this->unk266 != 0) {
+        this->unk266--;
+    }
```

Comment:
```suggestion
    } else {
        if (fabsf(player->actor.world.pos.y - this->actor.world.pos.y) < 200.0f) {
            func_801A1D44(&this->actor.projectedPos, 0x2E, 540.0f);
        }
        if (this->unusedTimer != 0) {
            this->unusedTimer--;
        }
        if (this->unk266 != 0) {
            this->unk266--;
        }
    }
```
would this work to get rid of the `return;`?